app/src/main/java/org/stepik/android/domain/course/interactor/CourseInteractor.kt

```diff
@@ -49,8 +58,9 @@ constructor(
         zip(
             courseStatsInteractor.getCourseStats(listOf(course)),
             solutionsInteractor.fetchAttemptCacheItems(course.id, localOnly = true),
-            if (promo == null) Single.just(PromoCode.EMPTY) else courseStatsInteractor.checkPromoCodeValidity(course.id, promo)
-        ) { courseStats, localSubmissions, promoCode ->
+            if (promo == null) Single.just(PromoCode.EMPTY) else courseStatsInteractor.checkPromoCodeValidity(course.id, promo),
+            if (sharedPreferenceHelper.authResponseFromStore != null) wishlistRepository.getWishlistRecord(DataSourceType.CACHE) else Single.just(StorageRecord(data = WishlistWrapper.EMPTY, kind = KIND_WISHLIST))
```

Comment:
let at least wrap `sharedPreferenceHelper.authResponseFromStore` in `Interactor`